code/modules/modular_computers/file_system/programs/system/file_browser.dm

```diff
@@ -182,45 +142,35 @@
 		var/datum/computer_file/F = HDD.find_file_by_name(href_list["PRG_copytousb"])
 		if(!F || !istype(F))
 			return 1
-		var/checkpass = ""
-		var/currentpass = F.password
 		var/datum/computer_file/C = F.clone(0)
-		if(!currentpass)
-			RHDD.store_file(C)
-		else
-			checkpass = sanitize(input(usr, "Please enter the password:"))
-			if(checkpass == currentpass)
-				RHDD.store_file(C)
-			else
-				return 1
+		RHDD.store_file(C)
 	if(href_list["PRG_copyfromusb"])
 		. = 1
 		var/obj/item/weapon/computer_hardware/hard_drive/HDD = computer.hard_drive
 		var/obj/item/weapon/computer_hardware/hard_drive/portable/RHDD = computer.portable_drive
 		if(!HDD || !RHDD || computer.enrolled != 2)
 			return 1
 		var/datum/computer_file/F = RHDD.find_file_by_name(href_list["PRG_copyfromusb"])
-		if(!F || !istype(F))
-			return 1
-		var/checkpass = ""
-		var/currentpass = F.password
 		if(!F || !istype(F))
 			return 1
 		var/datum/computer_file/C = F.clone(0)
-		if(!currentpass)
-			HDD.store_file(C)
-		else
-			checkpass = sanitize(input(usr, "Please enter the password:"))
-			if(checkpass == currentpass)
-				HDD.store_file(C)
-			else
-				return 1
+		HDD.store_file(C)
 	if(.)
 		SSnanoui.update_uis(NM)
 
 /datum/nano_module/program/computer_filemanager
 	name = "NTOS File Manager"
 
+/datum/computer_file/proc/can_acess_file(input_password = "")
+	if(!password)
+		return TRUE
+	else
+		input_password = sanitize(input(usr, "Please enter a password to access file '[filename]':"))
```

Comment:
Usage of `usr` outside of Topic and a few other procs is forbidden. It is unsafe and makes procs unusable. This is why you need the argument of `mob/user`.